### PR TITLE
add metrics reporting

### PR DESCRIPTION
**Pull request by lennessyy**

### CODE REVIEW

**Review by wenqilitg — APPROVED** [review]

File: modules/system-management/pages/management-commands.adoc [hunk 1]
```diff
@@ -552,6 +552,41 @@ ZK     : /home/tigergraph/tigergraph/log/zk/ZK#1.out
 ZK     : /home/tigergraph/tigergraph/log/zk/zookeeper.log
 ----
 
+=== gadmin metric
+
+This command reports information on resource consumption on a cluster or select nodes in a cluster, including CPU usage, memory usage, diskspace usage, and network usage.
```
Comment (a-m-thomas) [inline modules/system-management/pages/management-commands.adoc:557]:
```suggestion
This command reports information about resource consumption on a cluster or select nodes in a cluster, including CPU usage, memory usage, disk space usage, and network usage.
```

File: modules/system-management/pages/system-metrics.adoc [hunk 2]
```diff
@@ -0,0 +1,102 @@
+= Metrics Reporting
+:description: Overview of metrics gathering in TigerGraph Server.
+
+You can use `gadmin metric` commands to collect the following metrics related to state of resource usage on your instances:
+
+* CPU usage
+* Memory usage
+* Diskspace usage
```
Comment (a-m-thomas) [inline modules/system-management/pages/system-metrics.adoc:8]:
```suggestion
* Disk space usage
```

File: modules/system-management/pages/system-metrics.adoc [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+* Network-related information
+
+For a full list of options available to `gadmin metric`, run `gadmin metric -h` in the command line to view the help text for the command.
+
+== Report CPU usage
+
+Use command `gadmin metric -t cpu` to collect CPU usage by different TigerGraph services on a cluster or a node in the cluster.
```
Comment (a-m-thomas) [inline modules/system-management/pages/system-metrics.adoc:15]:
```suggestion
Use the command `gadmin metric -t cpu` to collect CPU usage by different TigerGraph services on a cluster or a node in the cluster.
```

Comment (a-m-thomas) [inline modules/system-management/pages/system-metrics.adoc:15]: Does it just write to console? "collect" sounds to me like it creates a log, how about "report"?

File: modules/system-management/pages/system-metrics.adoc [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+
+To report CPU usage on select nodes, use the `--host` flag or the `-m` shorthand, followed by the nodes for which you want CPU usage data, separated by commas.
+If you choose not to specify nodes, then the command reports CPU usage for all nodes in the cluster.
+
+To select specific services to report data for, specify the service IDs after the command.
+If you don't specify a service, the command reports data for all services
```
Comment (a-m-thomas) [inline modules/system-management/pages/system-metrics.adoc:21]:
```suggestion
If you don't specify a service, the command reports data for all services.
```

File: modules/system-management/pages/system-metrics.adoc [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+`gpe gse restpp gsql ifm gui ctrl kafka etcd zk nginx ts3 ts3serv dict admin kafkaconn kafkastrm-ll`
+
+The following command reports CPU usage by the `GUI` service for m1 and m2:
+
+[.wrap,console]
+----
+$ gadmin metric gui -t cpu -m m1,m2
+----
+
+The following command reports CPU usage by all services by m1 and m2:
```
Comment (a-m-thomas) [inline modules/system-management/pages/system-metrics.adoc:33]:
```suggestion
The following command reports CPU usage by all services for m1 and m2:
```

File: modules/system-management/pages/system-metrics.adoc [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+
+== Report memory usage
+
+Use command `gadmin metric -t mem` to report memory usage on a cluster or a node in the cluster.
+
+To report memory usage on select nodes, use the `--host` flag or the `-m` shorthand, followed by the nodes for which you want memory usage data.
+If you choose not to specify nodes, then the command reports memory usage for all nodes in the cluster.
+
+To select specific services to report data for, specify the service IDs after the command.
+If you don't specify a service, the command reports data for all services
```
Comment (a-m-thomas) [inline modules/system-management/pages/system-metrics.adoc:50]:
```suggestion
If you don't specify a service, the command reports data for all services.
```

File: modules/system-management/pages/system-metrics.adoc [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+The following command reports memory usage by all services for m1 and m2:
+
+[.wrap,console]
+----
+$ gadmin metric -t mem -m m1,m2
+----
+
+== Report disk space usage
+
+Use command `gadmin metric -t dsik` to report diskspace usage for a cluster or select nodes in a cluster.
```
Comment (a-m-thomas) [inline modules/system-management/pages/system-metrics.adoc:72]:
```suggestion
Use command `gadmin metric -t disk` to report disk space usage for a cluster or select nodes in a cluster.
```

File: modules/system-management/pages/system-metrics.adoc [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+The command reports three data points for each node:
+
+* `TCP CONNS`: number of open connections.
+* `IN`: Network inflow since the node started.
+* `OUT`: Network outflow since the node started.
+
+To report network information on select nodes, use the `--host` flag or the `-m` shorthand, followed by the nodes for which you want network data.
+If you choose not to specify nodes, then the command reports data for all nodes in the cluster.
+
+The following command reports network information by for m1 and m2:
```
Comment (a-m-thomas) [inline modules/system-management/pages/system-metrics.adoc:97]:
```suggestion
The following command reports network information for m1 and m2:
```

File: modules/system-management/pages/system-metrics.adoc [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+
+[.wrap,console]
+----
+$ gadmin metric -t disk -m m1,m2
```
Comment (a-m-thomas) [inline modules/system-management/pages/system-metrics.adoc:101]:
```suggestion
$ gadmin metric -t net -m m1,m2
```

**Review by alexthomasTG — CHANGES_REQUESTED** [review]

**Review by alexthomasTG — APPROVED** [review]

File: modules/system-management/pages/system-metrics.adoc [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+:description: Overview of metrics gathering in TigerGraph Server.
+
+You can use `gadmin metric` commands to collect the following metrics related to state of resource usage on your instances:
+
+* CPU usage
+* Memory usage
+* Disk space usage
+* Network-related information
+
+For a full list of options available to `gadmin metric`, run `gadmin metric -h` in the command line to view the help text for the command.
```
Comment (mbmurrell) [inline modules/system-management/pages/system-metrics.adoc:11]: We should say this, however, the wording makes it seem as though we are not listing the full list of metrics available. A complete list needs to be provided in the documentation

File: modules/system-management/pages/system-metrics.adoc [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+$ gadmin metric gui -t cpu -m m1,m2
+----
+
+The following command reports CPU usage by all services for m1 and m2:
+
+[.wrap,console]
+----
+$ gadmin metric gui -t cpu -m m1,m2
+----
+
```
Comment (mbmurrell) [inline modules/system-management/pages/system-metrics.adoc:39]: Do we have a master table of all gadmin commands and their function? If so, we need to add a link to it. If not, then we should consider for future work.

**Review by MaryBeth-Murrell — APPROVED** [review]

File: modules/system-management/pages/system-metrics.adoc (hunk 11 above)
Comment (lennessyy) [reply modules/system-management/pages/system-metrics.adoc:39]: A master list of all the commands are here: https://docs.tigergraph.com/tigergraph-server/current/system-management/management-commands

Though some of these are likely out-of-date, and the formatting could be improved as well.

File: modules/system-management/pages/system-metrics.adoc (hunk 10 above)
Comment (lennessyy) [reply modules/system-management/pages/system-metrics.adoc]: Linked to the reference for the command

**Pull request merged**